src/main/webapp/WEB-INF/templates/system/common/ice.ftl

```diff
@@ -370,3 +370,113 @@ Crafter CMS Authoring Scripts
     </#if>
   </@tag>
 </#macro>
+
+<#macro navigation
+<#---->url="/"
+<#---->omitNavElement=false
+<#---->containerEl="ul"
+<#---->itemWrapperEl="li"
+<#---->itemWrapperClass=""
+<#---->itemWrapperActiveClass="active"
+<#---->itemWrapperAttributes={}
+<#---->itemClass=""
+<#---->itemActiveClass="active"
+<#---->itemAttributes={}
+<#---->deepItemClass=""
+<#---->deepItemClassPrefix=""
+<#---->deepItemWrapperClassPrefix=""
+<#---->depth=1
+<#---->includeRoot=true
+<#---->inlineHomeWithImmediateChildren=true
+>
+  <#assign navTree = navTreeBuilder.getNavTree(url, depth, Request.pageUrl)/>
+  <#if omitNavElement != false><nav></#if>
+  <#if containerEl != ""><${containerEl}></#if>
+    <@navigationItem
+      containerEl=containerEl
+      itemWrapperEl=itemWrapperEl
+      itemWrapperClass=itemWrapperClass
+      itemWrapperActiveClass=itemWrapperActiveClass
+      itemWrapperAttributes=itemWrapperAttributes
+      itemClass=itemClass
+      itemActiveClass=itemActiveClass
+      itemAttributes=itemAttributes
+      deepItemClass=deepItemClass
+      deepItemClassPrefix=deepItemClassPrefix
+      deepItemWrapperClassPrefix=deepItemWrapperClassPrefix
+      depth=depth
+      currentDepth=0
+      navItem=navTree
+      includeRoot=includeRoot
+      inlineHomeWithImmediateChildren=inlineHomeWithImmediateChildren
+    />
+  <#if containerEl != ""></${containerEl}></#if>
+  <#if omitNavElement != false></nav></#if>
+</#macro>
+
+<#macro navigationItem
+<#---->containerEl="ul"
+<#---->itemWrapperEl="li"
+<#---->itemWrapperClass=""
+<#---->itemWrapperActiveClass="active"
+<#---->itemWrapperAttributes={}
+<#---->itemClass=""
+<#---->itemActiveClass="active"
+<#---->itemAttributes={}
+<#---->deepItemClass=""
+<#---->deepItemClassPrefix=""
+<#---->deepItemWrapperClassPrefix=""
+<#---->depth=1
+<#---->currentDepth=0
+<#---->navItem={}
+<#---->includeRoot=true
+<#---->inlineHomeWithImmediateChildren=true
+>
+  <#if itemWrapperEl != "">
+    <${itemWrapperEl}
+    class="${navItem.active?then(itemWrapperActiveClass, '')} ${itemWrapperClass}"
+      <#list itemWrapperAttributes as attr, value>
+          ${attr}="${value}"
+      </#list>
+    >
+  </#if>
+    <#assign storeUrl = urlTransformationService.transform('renderUrlToStoreUrl', navItem.url)>
+    <#assign item = siteItemService.getSiteItem(storeUrl) />
```

Comment:
@avasquez614 @sumerjabri would this be a performance concern?
ICE requires model information so we need to fetch the model for each printed nav item.

Reply:
If so, is there some sort of bulk call we could perform to retrieve all pages at once instead of fetching per iteration?